src/tests/Microsoft.Diagnostics.NETCore.Client/Microsoft.Diagnostics.NETCore.Client.UnitTests.csproj

```diff
@@ -1,5 +1,41 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
+  <Target Name="_generateDARCVersionCs"
+          Inputs="$(VersionsPropsPath)"
+          Outputs="$(IntermediateOutputPath)/DARCVersion.g.cs">
+    <PropertyGroup>
+      <DARCVersionFile>
+<![CDATA[
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+// THIS FILE IS AUTO-GENERATED DURING BUILD.  MODIFICATIONS _WILL_ BE OVERWRITTEN!
+
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    public partial class CommonHelper
+    {
+        public static readonly string CurrentDARCVersion = "$(MicrosoftNETCoreAppVersion)";
+    }
+}
+]]>
+      </DARCVersionFile>
+    </PropertyGroup>
+
+    <WriteLinesToFile File="$(IntermediateOutputPath)/DARCVersion.g.cs"
+                      Overwrite="true"
+                      Lines="$(DARCVersionFile.Replace(';', '%3B'))"
+                      WriteOnlyWhenDifferent="true" />
+    <ItemGroup>
+      <FileWrites Include="$(IntermediateOutputPath)/DARCVersion.g.cs" />
+      <Compile Include="$(IntermediateOutputPath)/DARCVersion.g.cs" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="InjectDARCVersion" BeforeTargets="BeforeCompile" DependsOnTargets="_generateDARCVersionCs" />
```

Comment:
The snippet
```xml
    <ItemGroup>
      <FileWrites Include="$(IntermediateOutputPath)/DARCVersion.g.cs" />
      <Compile Include="$(IntermediateOutputPath)/DARCVersion.g.cs" />
    </ItemGroup>
```
should be in this target. Even if you don't write a file in the incremental case (you've built before and you don't change the RT version file), you need to add the file to the compile group, otherwise the inputs to the compiler have changed and you no longer have the variable and things will fail to compile.